components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -543,18 +542,11 @@ impl CanvasRenderingContext2D {
                         Some(ref canvas) => &**canvas,
                     };
 
-                    let window = window_from_node(canvas);
+                    let style = canvas.upcast::<Element>().style();
 
-                    let style = window.GetComputedStyle(canvas.upcast(), None);
-
-                    let element_not_rendered =
-                        !canvas.upcast::<Node>().is_in_doc() ||
-                        style.GetPropertyValue(DOMString::from("display")) == "none";
-
-                    if element_not_rendered {
-                        Ok(RGBA::new(0, 0, 0, 255))
-                    } else {
-                        self.parse_color(&style.GetPropertyValue(DOMString::from("color")))
+                    match style {
+                        Some(ref s) if canvas.upcast::<Node>().is_in_doc() => Ok(s.get_color().color),
```

Comment:
This also needs to check whether `s.get_box().display` is not `Display::None`.

Reply:
Also, you should be able to assert that `is_in_doc()` if the style is `Some(..)`. But checking it explicitly doesn't hurt I guess.

Reply:
Thanks emilio! I realised about `s.get_box().display` from the error I made previously. Was wondering about `is_in_doc()` as well but wasn't too sure & decided to spend some time reading up.